### ISSUE TITLE
Observe idle state in reader client

### ISSUE THREAD

**Issue by EronWright** [pravega/pravega#1285]
**Problem description**
To implement watermark idleness in Flink, the connector must detect when it has no segments assigned.   Make segment assignment sufficiently observable to facilitate that.

Note that watermark idleness is unrelated to broader event time support.

**Problem location**
EventStreamReader

**Suggestions for an improvement**
Consider extending `EventRead` to convey entering an idle state (i.e. where no segments are assigned).   This type of event would be similar to a checkpoint event.   The caller would mark itself as being in an idle state then continue to call `readNextEvent` which would block as normal.

**Comment by tkaitchuck** [comment]
I don't think this requires a new API. I think that https://github.com/pravega/flink-connectors/issues/17 is mistaken. Downstream processes need to see watermarks at regular intervals not just when there are no segments, but also when there is no data on a segment. Consider the case where there are 10 segments and for a very long period only one has any data, to keep processing so if there are 5 readers, then we need all of them to emit watermarks so that downstream workers can make progress. 
So the solution in the connector is to call `event = readNextEvent(5000)` that will return every 5 seconds, if the `event.getEvent()`  returns `null` then there is no data and a watermark can be emitted.

**Comment by fpj** [comment]
This issue will be resolved once we complete the watermarking implementation.

**Comment by vijikarthi** [comment]
> So the solution in the connector is to call event = readNextEvent(5000) that will return every 5 seconds, if the event.getEvent() returns null then there is no data and a watermark can be emitted.

I am not sure if this is the correct way to indicate the advancement of time to the downstream operators unless the watermark is asserted by the source (Pravega) itself. An alternate option is to get the position information from the `EventRead` and see if the reader owns any segments? If not, then we could mark the reader as temporarily idle and Flink will ignore the source operator from watermark consideration.

```
PositionImpl positionImpl = (PositionImpl)eventRead.getPosition();
if (positionImpl.getOwnedSegments().isEmpty() && event == null) {
     ctx.markAsTemporarilyIdle();
 }
```

**Comment by shiveshr** [comment]
Closing the issue as Pravega Watermarks are implemented #3344
